Kernel/Bus/USB/Drivers/MassStorage/MassStorageDriver.cpp

```diff
@@ -54,24 +38,43 @@ ErrorOr<void> MassStorageDriver::probe(USB::Device& device)
         // FIXME: There might be multiple MassStorage configs present,
         //        figure out how to decide which one to take,
         //        although that's very unlikely
-        bool has_accepted_an_interface = false;
+        Optional<USBInterface const&> bot_interface;
+        Optional<USBInterface const&> uas_interface;
+
         for (auto const& interface : config.interfaces()) {
-            // FIXME: Handle multiple interfaces
-            //        Interface may coexist at the same time,
-            //        but having multiple handles on the same data storage seems like a bad idea,
-            //        so:
-            // FIXME: Choose the best supported interface
-            //        UAS for example is supposed to be better than BBB, but BBB will always
-            //        be the first listed interface of them, when both are supported
-            auto result = checkout_interface(device, interface);
-            if (result.is_error())
+            if (interface.descriptor().interface_class_code != USB_CLASS_MASS_STORAGE)
                 continue;
 
-            has_accepted_an_interface = true;
+            if (interface.descriptor().interface_protocol == to_underlying(TransportProtocol::UAS))
+                uas_interface = interface;
+            else if (interface.descriptor().interface_protocol == to_underlying(TransportProtocol::BBB))
+                bot_interface = interface;
+            else
+                dmesgln("USB MassStorage Interface for device {:04x}:{:04x} has unsupported protocol {}", device.device_descriptor().vendor_id, device.device_descriptor().product_id,
+                    transport_protocol_string(static_cast<TransportProtocol>(interface.descriptor().interface_protocol)));
+        }
+
+        if (!bot_interface.has_value() && !uas_interface.has_value())
+            continue;
+
+        dmesgln("USB MassStorage Interfaces for device {:04x}:{:04x} found:", device.device_descriptor().vendor_id, device.device_descriptor().product_id);
+        dmesgln("    Configuration: {}", config.configuration_id());
+        dmesgln("    BOT Interface: {}", bot_interface.has_value());
+        dmesgln("    UAS Interface: {}", uas_interface.has_value());
+
+        if (uas_interface.has_value() && device.speed() != USB::Device::DeviceSpeed::SuperSpeed) {
+            // FIXME: We only support UAS on version < 3.0 devices
+            //        as we don't support waiting ERDY transactions yet
```

Comment:
These references to ERDY should probably replaced with streams (also in the commit message).